units/stats_amboy.go

```diff
@@ -83,11 +83,26 @@ func (j *amboyStatsCollector) Run(ctx context.Context) {
 		})
 	}
 
-	remoteQueue := j.env.RemoteQueue()
-	if !j.ExcludeRemote && (remoteQueue != nil && remoteQueue.Info().Started) {
-		j.logger.Info(message.Fields{
-			"message": "amboy remote queue stats",
-			"stats":   remoteQueue.Stats(ctx),
-		})
+	if !j.ExcludeRemote {
+		remoteQueue := j.env.RemoteQueue()
+		if remoteQueue != nil && remoteQueue.Info().Started {
+			j.logger.Info(message.Fields{
+				"message": "amboy remote queue stats",
+				"stats":   remoteQueue.Stats(ctx),
+			})
+		}
+
+		for _, queueGroupName := range j.env.RemoteQueueGroup().Queues(ctx) {
+			queueStatsByQueue := map[string]amboy.QueueStats{}
+			queueGroup, _ := j.env.RemoteQueueGroup().Get(ctx, queueGroupName)
```

Comment:
We can't `Get` all the queues in the queue group because the queue group queues have a TTL where they get pruned (i.e. stop processing) after some amount of time. `Get` refreshes that TTL, which would prevent it from pruning temporary queues, like the per-version queues created by `generate.tasks`. It's also the case that `Stats` on the queue group is less efficient than `Stats` on the general remote queue, so issuing too many of these queries frequently can be expensive.

If possible, just doing this for the important queues of interest that never get TTL'd (like event notifier and host termination) seems safer.

Reply:
I had a feeling this wasn't going to work for some reason :llama-distraught: thank you for the detailed explanation! Let's just go with the host termination queue for now then instead of making this general; planning to leave this in this job but lmk if that's not what you were envisioning